ThermofluidStream/Processes/FlowResistance.mo

```diff
@@ -5,23 +5,25 @@ model FlowResistance "Flow resistance model"
     final clip_p_out=true);
 
   import Modelica.Constants.pi "Constant Pi";
+  import ThermofluidStream.Processes.Internal.ShapeOfResistance "Shape of cross sectional area";
 
-  parameter Internal.ShapeOfResistance shape=ThermofluidStream.Processes.Internal.ShapeOfResistance.circular
+  parameter SI.Length l(min=0) "Length of pipe"
+    annotation (Dialog(group = "Geometry"));
+  parameter ShapeOfResistance shape=ShapeOfResistance.circular
     "Shape of cross sectional area"
     annotation (Dialog(group = "Geometry",enable = true),
     choices(
       choice=ThermofluidStream.Processes.Internal.ShapeOfResistance.circular "Circular",
       choice=ThermofluidStream.Processes.Internal.ShapeOfResistance.rectangle "Rectangle",
       choice=ThermofluidStream.Processes.Internal.ShapeOfResistance.other "Other"));
 
-  parameter SI.Length l(min=0) "Length of pipe" annotation (Dialog(group = "Geometry"));
   parameter SI.Radius r(min=0) = 0 "Radius of pipe"
   annotation (Dialog(group = "Geometry", enable=(shape == ThermofluidStream.Processes.Internal.ShapeOfResistance.circular)));
   parameter SI.Length a(min=0) = 0 "Rectangle width"
     annotation(Dialog(group = "Geometry", enable=(shape == ThermofluidStream.Processes.Internal.ShapeOfResistance.rectangle)));
   parameter SI.Length b(min=0) = 0 "Rectangle height"
     annotation(Dialog(group = "Geometry", enable=(shape == ThermofluidStream.Processes.Internal.ShapeOfResistance.rectangle)));
-  parameter SI.Area areaCross = 0 "Cross section area"
+  parameter SI.Area areaCross = areaCrossActual "Cross section area"
```

Comment:
This is not good for a) perpetual dependency of `areaCross` and `areaCrossActual` if shape <> circular AND shape <> rectangle (i.e. if shape == other, see equation for `areaCrossActual`) and b) dependency of a "public" parameter on a "private" one makes it IMO harder for user to follow.
So probably best to set value zero (0) here (similar to `r`, `a`, ...).

Additionally, the modifier `min=0` shall be probably set for `areaCross` as well.

Reply:
I agree here. My intention was, that the resulting cross-sectional area can be obtained in the simulation results without introducing an extra parameter. If I put `areaCross = 0` per default, this will show up as 0 in the results and the user might wonder...

Reply:
Options:
- rename the parameter `areaCross` and/or  make `areaCrossActual` public (probably not good).
- add annotation `HideResult = true` to `areaCross` ("[it] defines that the model developer proposes to not show the simulator results of the corresponding component [e.g., it will not be possible to plot this variable].) But I don't know if this really is desired behavior.

Reply:
I made all parameters public now because they are of interest for the modeller anyways. Like that it is also easy to compare the cross-section area of the component against the resulting hydraulic cross-sectional area, if the shape is not circular. For inertance calculation the hydraulic cross-section is taken.